File: src/main/java/me/leodev/trollingfreedomreborn/main/TrollGUIAlias.java
package me.leodev.trollingfreedomreborn.main;

import me.leodev.trollingfreedomreborn.ui.PlayerSelectorInventory;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;


//ArgsCmd also implements CommandInterface
public class TrollGUIAlias implements CommandExecutor {

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage("§cOnly players can use the GUI.");
            return true;
        }

        Player p = (Player) sender;

        if (!p.hasPermission("trollingfreedom.open")) {
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', (String) Core.instance.getConfig().get("no-perms")));
            return true;
        }

        // If they typed a name: /troll Technoblade
        if (args.length > 0) {
            Player target = Bukkit.getPlayer(args[0]);
            if (target != null) {
                // THE CRITICAL CHECK
                if (!Core.canTroll(target)) {
                    p.sendMessage("§c§l[TFR] §7This player is §4protected §7and cannot be trolled.");
                    return true;
                }
                // If not blocked, open the specific troll menu for that target
                // (Assuming your UI has a way to open directly to a player's page)
            }
        }

        // Default: Open the player selector
        new PlayerSelectorInventory().openSel(p);
        return true;
    }

}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/main/TrollGUIAlias.kt
package com.leomadrassi.trollingfreedomreborn.main

import com.leomadrassi.trollingfreedomreborn.ui.PlayerSelectorInventory
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class TrollGUIAlias : CommandExecutor {

    override fun onCommand(sender: CommandSender, cmd: Command, commandLabel: String, args: Array<String>): Boolean {
        if (sender !is Player) {
            sender.sendMessage("§cOnly players can open the GUI.")
            return true
        }

        val p = sender

        if (!p.hasPermission("trollingfreedom.open")) {
            sender.sendMessage(ChatColor.translateAlternateColorCodes('&', Core.instance!!.pluginConfig.get("no-perms") as String))
            return true
        }

        if (args.isNotEmpty()) {
            val target = Bukkit.getPlayer(args[0])
            if (target != null) {
                if (!Core.canTroll(target)) {
                    p.sendMessage("§c§l[TFR] §7That player is §4blocked §7from trolling.")
                    return true
                }
            }
        }

        PlayerSelectorInventory().openSel(p)
        return true
    }
}
